test/cpp/src/TestClient.cpp

```diff
@@ -488,7 +488,7 @@ int main(int argc, char** argv) {
       testClient.testBinary(bin_result, string(bin_data, 256));
       if (bin_result.size() != 256) {
         printf("}\n*** FAILED ***\n");
-        printf("invalid length: %lu\n", bin_result.size());
+        printf("invalid length: %zu\n", bin_result.size());
```

Comment:
Yeah +1, we could get rid of the printf's all together... 
The boost format is cool, but I thought we're looking into getting rid of it too? hehe